src/main/xar-resources/modules/search.xql

```diff
@@ -20,54 +22,142 @@ declare variable $dq:CHARS_KWIC := 80;
     Templating function: process the query.
 :)
 declare 
-    %public %templates:default("field", "all") %templates:default("view", "summary")
-function dq:query($node as node()*, $model as map(*), $q as xs:string?, $field as xs:string, $view as xs:string) {
+    %public
+    %templates:default("field", "all")
+    %templates:default("view", "summary")
+function dq:query($node as node()*, $model as map(*), $q as xs:string?, $field as xs:string, $view as xs:string) as element(div)? {
 	if ($q) then
 		let $hits := dq:do-query(collection($config:data-root), $q, $field)
-		let $search-params :=
-            string-join(
-                map-pairs(function($k, $v) { $k || "=" || $v }, ("q", "field"), ($q, $field)),
-                "&amp;"
-            )
 		return
             <div id="f-search">
-			{dq:print-results($hits, $search-params, $view)}
+			{
+                dq:print-results($hits, map { "q": $q, "field": $field}, $view)
+			}
             </div>
 	else
 		()
 };
 
+(:~
+: Returns the elements for which the $query matches.
+:
+: @param context the nodes to search
+: @param query the full-text query
+: @param field the name of a field, if the query should be restricted to a specific field
+:
+: @return The elements that match the query, typically one of:
+:     db5:title, db5:keyword, db5:para, db5:sect1, db5:sect2, db5:sect3. 
+:)
+declare
+    %public
+function dq:do-query($context as node()*, $query as xs:string?, $field as xs:string?) as element()* {
+    switch ($field)
+        case "title" return
+            $context/db5:article/db5:info/db5:title[ft:query(., $query)]
+            |
```

Comment:
I haven't seen this syntax before
```xquery
return
  $a
  |
  $b
```
 is this a variant of
```xquery
return
  ($a, $b)
```
?

Reply:
`|` is a `Union`. See https://www.w3.org/TR/xquery-31/#doc-xquery31-UnionExpr

`($a, $b)` is a computed sequence constructor for a sequence of `$a` and `$b`